build/ci/github_actions/operator/wait_for_driver_ready.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash -xe
+set +o pipefail
+
+driver_is_ready=false
+actual_driver_running_time_in_seconds=0
+minimum_driver_running_time_in_seconds=10
+containers_suffix=ibm-block-csi
+declare -a driver_pods_types=(
+  "controller"
+  "node"
+)
+
+get_csi_pods (){
+  kubectl get pod -A -l csi
+}
+
+get_image_pod_by_type (){
+  pod_type=$1
+  container_to_check=$2
+  containers_images=`kubectl get pods $(get_csi_pods | grep $pod_type | awk '{print$2}') -o jsonpath='{range .spec.containers[*]}{.name},{.image} {end}'`
+  for containers_image in $containers_images
+  do
+    if [[  "$containers_image" =~ "$container_to_check," ]]; then
+      echo $containers_image | awk -F , '{print$2}'
+      break
+    fi
+  done
+}
+
+wait_for_driver_pod_to_start (){
+  driver_pod_type=$1
+  while [ "$(get_csi_pods | grep $driver_pod_type | wc -l)" -eq 0 ]; do
+    echo "The $driver_pod_type is not deployed"
+    sleep 1
+  done
+}
+
+wait_for_driver_deployment_to_start (){
+  for driver_pods_type in "${driver_pods_types[@]}"
+  do
+    wait_for_driver_pod_to_start $driver_pods_type
+  done
+}
+
+wait_for_driver_deployment_to_finish (){
+  while [ $driver_is_ready == "false" ]; do
+    if [ "$(get_csi_pods | grep -iv running | grep -iv name | wc -l)" -eq 0 ]; then
+      ((++actual_driver_running_time_in_seconds))
+      if [ $actual_driver_running_time_in_seconds -eq $minimum_driver_running_time_in_seconds ]; then
+        driver_is_ready=true
+      fi
+    else
+      actual_driver_running_time_in_seconds=0
+    fi
+    get_csi_pods
+    sleep 1
+  done
+}
+
+assert_expected_image_in_pod (){
+  pod_type=$1
+  expected_pod_image=$2
+  container_to_check=$containers_suffix-$pod_type
+  image_in_pod=`get_image_pod_by_type $pod_type $container_to_check`
+  if [[ $image_in_pod != $expected_pod_image ]]; then
+    echo "$pod_type's image ($image_in_pod) is not the expected image ($expected_pod_image)"
+    exit 1
+  fi
+}
+
+assert_pods_images (){
+  expected_node_image=$node_repository_for_test:$driver_images_tag
+  expected_controller_image=$controller_repository_for_test:$driver_images_tag
+  expected_operator_image=$operator_image_repository_for_test:$operator_specific_tag_for_test
```

Comment:
shouldn't this be checked before the cr is even applied?
this filename suggests that this step is for the driver

Reply:
> > shouldn't this be checked before the cr is even applied?
> 
> I think it is better to check it here because if we check it before it has been applied, it will check my script and not the operator.
> When we are checking it here we are checking if the operator is really applying the images in the cr

I don't see how checking the operator image is related to applying the cr

Reply:
oh, I thought you talked about the controller and the node as well.
So I don't think it really matters where you are checking the image but if we already do image checking here to the controller and to the node, I think it better to check the operator also here.

Reply:
if we want to check the operator image (and I think we do), I don't see a reason to delay it until after applying the cr (there's no need to even apply the cr if the operator image is wrong)

Reply:
but the best check if the right operator image tag is deployed, is only after the deployment of the operator.
And the thing that you suggesting is to check the script code

Reply:
yes, I think we should check the operator image after the operator deployment and before applying the cr.
checking the script here is a good idea, especially since it currently uses `sed`

Reply:
I don't think we should check the script, if I checked it before it will work, and if it won't we will see it in the deployment, why checking the script?

Reply:
you only checked it once in the current state of the code.
how would we see it in the deployment? I think the test can just pass without us knowing that the relevant operator code wasn't tested

Reply:
e.g. if tomorrow we stop using `ibmcom`, the sed won't have any effect, and we could have a false positive pass

Reply:
> how?

e.g. if sed changes in a way that fails to replace the image

Reply:
> so if this will be the case we will see it here because the operator will running with wrong image

so if you agree that the test should check the operator image, why wait until after applying the cr?

Reply:
> > how?
> 
> e.g. if the `sed` command changes in a way that fails to replace the image

great, so you will see it in the operator+driver deployment

Reply:
> > so if this will be the case we will see it here because the operator will running with wrong image
> 
> so if you agree that the test should check the operator image, why wait until after applying the cr?

because I don't want to check the same thing twice, it does not cost you to deploy them with a bad image and I don't want to check the script, there is a lot of places that you do stuff, we can't check every step, what is the difference in this change that the one in controller and node?

Reply:
as per our conversation, I meant to check the operator pod, not yaml, and only once, before applying the cr

Reply:
done